contracts/src/ethereum/TokenBridge.sol

```diff
@@ -143,17 +143,34 @@ contract TokenBridge is
         messagingContract.consumeMessageFromL2(l2TokenBridge, payload);
     }
 
-    function deposit(address l1Token_, uint256 l2Recipient, uint256 amount) isApprovedToken(l1Token_) external {
-        IStaticATokenLM l1Token = IStaticATokenLM(l1Token_);
+    function deposit(IERC20 l1Token, uint256 l2Recipient, uint256 amount) external {
         l1Token.transferFrom(msg.sender, address(this), amount);
-        sendMessage(l1Token_, l2Recipient, amount);
+        sendMessage(address(l1Token), l2Recipient, amount);
     }
-    
-    function withdraw(address l1Token_, address recipient, uint256 amount) isApprovedToken(l1Token_) external {
-        consumeMessage(l1Token_, recipient, amount);
-        require(recipient != address(0x0), "INVALID_RECIPIENT");
-        IStaticATokenLM l1Token = IStaticATokenLM(l1Token_);
-        require(l1Token.balanceOf(msg.sender) - amount <= l1Token.balanceOf(msg.sender), "UNDERFLOW");
+
+    function depositUnderlying(
+        IStaticATokenLM l1Token,
+        uint256 l2Recipient,
+        uint256 amount,
+        uint16 refferalCode,
+        bool fromAsset
+    ) external {
+        amount = l1Token.deposit(msg.sender, amount, refferalCode, fromAsset);
```

Comment:
we could add comments to clarify the behavior of the two `deposit` functions:
- above this line you could add: `// deposit tokens or aTokens to get staticATokens`
- above the line below: `// transfer staticATokens to the bridge`